src/components/MainContainer.js

```diff
@@ -17,27 +17,35 @@ class MainContainer extends Component {
             dataset: {},
             dataLoaded: false, 
             geoid: defaultGeoid, 
+            STATS: [],
             graphW: 0,
             graphH: 0,
             mapContainerW: 0,
             mapContainerH: 0
         };
     };
 
-    componentDidMount() {
+    async componentDidMount() {
         window.addEventListener('resize', this.updateGraphDimensions);
         window.addEventListener('resize', this.updateMapContainerDimensions);
 
         this.updateGraphDimensions();
         this.updateMapContainerDimensions();
-        
+
         const { geoid } = this.state;
-        fetchJSON(geoid)
-            .then(dataset => this.setState({dataset}))
-            .catch(e => console.log('Fetch was problematic: ' + e.message))
-            .then(() => this.setState({dataLoaded: true}));
+        try {
+            const dataset = await fetchJSON(geoid);
+            const outcomesRaw = await fetchJSON('outcomes');
+            const outcomes = Object.keys(outcomesRaw).map((obj) => outcomesRaw[obj]);
+
+            this.setState({dataset, STATS: outcomes});
+            this.setState({dataLoaded: true});
```

Comment:
Note about react:

Even though these two setState calls look synchronous, they actually get optimized by React and get called together (after being merged in order) at the end of certain methods. I believe this is only true for React lifecycle methods (like componentDidMount) and React event handlers (like a Button component's onClick0

```
this.setState({dataset, STATS: outcomes});
this.setState({dataLoaded: true});
```

is also

```
this.setState({dataset, STATS: outcomes, dataLoaded: true});
```


https://stackoverflow.com/a/48610973/3361618

Reply:
@BenjiLee I know if I was using `.then` then I could just put it in another `.then` block that will execute after the first `.then`. However, in this case, I should be able to add `this.setState({dataLoaded: true});` into a `finally` clause, correct?

```
try {
            const dataset = await fetchJSON(geoid);
            const outcomesRaw = await fetchJSON('outcomes');
            const outcomes = Object.keys(outcomesRaw).map((obj) => outcomesRaw[obj]);

            this.setState({dataset, STATS: outcomes});
        } catch (e) {
            console.log('Fetch was problematic: ' + e.message)
        } finally {
            this.setState({dataLoaded: true});
        }
```